### DISCUSSION TITLE
How do I extract the individual hsv values?(urgent)

### DISCUSSION THREAD

**Discussion by sofalights** [pybricks/support#1920] [Q&A] [answered]
```
left_color_sensor = ColorSensor(Port.B)
right_color_sensor = ColorSensor(Port.F)

h_left, s_left, v_left = left_color_sensor.hsv(surface=True)
h_right, s_right, v_right = right_color_sensor.hsv(surface=True)
```
This is the code I'm using to extract the h,s, and v into its own little variables. 
However, the following error occurs:
```
Traceback (most recent call last):
  File "getshit.py", line 15, in <module>
TypeError: 'Color' object isn't iterable
```

How do I then extract the individual values? Would it work on a 2 sensor system?
Thanks!

**Comment by BertLindeman** [comment]
Hello,


According to the doc on the stable version of pybricks the `color_sensor.hsv` returns a `Color` value.

In the stable 3.5.0 version I got the same error as you describe.

Also tested the current beta firmware, no problem there.

You can go to beta.pybricks.com and install the firmware there and run the program.

I do not have two color sensors, so I ran this program one sensor at a time:
```python

from pybricks.pupdevices import ColorSensor
from pybricks.parameters import Port
from pybricks import version

print(version)

LEFT = False
if LEFT:
    left_color_sensor = ColorSensor(Port.B)
    h_left, s_left, v_left = left_color_sensor.hsv(surface=True)
    print(h_left, s_left, v_left)
else:
    right_color_sensor = ColorSensor(Port.A)
    h_right, s_right, v_right = right_color_sensor.hsv(surface=True)

    print(h_right, s_right, v_right)
```
With this result:
```
('technichub', '3.6.0b2', 'ci-release-81-v3.6.0b2 on 2024-10-15')
330 64 1
```

Bert

**Reply by sofalights** [reply]
I appreciate your help, however, even when I ran your code on beta.pybricks.com, the same error occurred. The maintainer's method did work on stable though, and as such his answer was accepted. Despite this, I sincerely thank you for helping me.

**Comment by laurensvalk** [comment, marked as answer]
You can do it as follows:

```python
color = left_color_sensor.hsv(surface=True)

color.h # this is the hue, and so on.
```

In the next release, iterating like you tried will also be supported.

**Reply by sofalights** [reply]
Thank you it works!

**Reply by laurensvalk** [reply]
Glad to hear it. Can we see the end result :smile:?

**Reply by sofalights** [reply]
sure!
code:
```
senB = ColorSensor(Port.B)
senF = ColorSensor(Port.F)

colorB = senB.hsv()
colorF = senF.hsv()

print("new reading below")
print(colorB,'left_sensor')
print(colorF,'right_sensor')
print(colorB.h,colorB.s,colorB.v,' left sens')
print(colorF.h,colorF.s,colorF.v,' right sens')
```

Result:
```
new reading below
Color(h=152, s=57, v=52) left_sensor
Color(h=153, s=55, v=52) right_sensor
152 57 52  left sens
153 55 52  right sens
```
